### PR TITLE
Support pytorch_optimizer

### PR DESCRIPTION
**Pull request by gesen2egee**
https://github.com/kozistr/pytorch_optimizer

A very practical optimizer library that includes many newer optimizer (or learning rate scheduling) options.

For example, to use CAME:
`--optimizer_type "pytorch_optimizer.CAME" --optimizer_args "weight_decay=0.01"
`
To use Lion with cautious mode:
```
--optimizer_type "pytorch_optimizer.Lion" --optimizer_args "weight_decay=0.01" "cautious=True"
You can check pytorch_optimizer to see what other options are available.
```

The author kozistr is very active. If you have an optimizer you’d like to try, 
you can provide a paper link or code and open a issue—he implements things quickly!

### CODE REVIEW

**Comment by 6DammK9** [comment]
FYI: This had covered #1409 already. Meanwhile this gives a lot more optimizers to explore other than `bitsandbytes`.
_If you experienced OOM on FFT (Full FineTuning), this may help (e.g. adafactor / LION / CAME / mini adam etc.)_

```py
    parser.add_argument(
        "--optimizer_type",
        type=str,
        default="",
        help="Optimizer to use / オプティマイザの種類: AdamW (default), AdamW8bit, PagedAdamW, PagedAdamW8bit, PagedAdamW32bit, "
        "Lion8bit, PagedLion8bit, Lion, SGDNesterov, SGDNesterov8bit, "
        "DAdaptation(DAdaptAdamPreprint), DAdaptAdaGrad, DAdaptAdam, DAdaptAdan, DAdaptAdanIP, DAdaptLion, DAdaptSGD, "
        "AdaFactor. "
        "Also, you can use any optimizer by specifying the full path to the class, like 'bitsandbytes.optim.AdEMAMix8bit' or 'bitsandbytes.optim.PagedAdEMAMix8bit'.",
    )
```

File: requirements.txt [hunk 1]
```diff
@@ -43,4 +42,7 @@ rich==13.7.0
 # for T5XXL tokenizer (SD3/FLUX)
 sentencepiece==0.2.0
 # for kohya_ss library
+pytorch-optimizer
```
[the remark this answers is not in the document]
Comment (gesen2egee) [reply requirements.txt:45]: okay
But this is a rapidly updating library.🤣
